app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -313,6 +323,55 @@ class BrowserFragment :
         }
 
         setSitePermissions(view)
+        showCfrForShieldToolbarIcon()
+    }
+
+    private fun showCfrForShieldToolbarIcon() {
+        val shieldToolbarIcon = binding.browserToolbar.rootView.findViewById<View>(
+            R.id.mozac_browser_toolbar_tracking_protection_indicator
+        )
+        if (shieldToolbarIcon != null && getCfrVisibilityForShieldToolbarIcon() &&
+            !tab.content.securityInfo.secure
+        ) {
+            _toolbarShieldIconCfrBinding = ToolbarShieldIconCfrBinding.inflate(LayoutInflater.from(context))
+            val toolbarShieldIconCfrPopupWindow =
+                PopupWindow(
+                    toolbarShieldIconCfrBinding.root,
+                    LinearLayout.LayoutParams.WRAP_CONTENT,
+                    LinearLayout.LayoutParams.WRAP_CONTENT,
+                    true
+                )
+
+            toolbarShieldIconCfrPopupWindow.showAsDropDown(
+                shieldToolbarIcon, 0,
+                TypedValue.applyDimension(
+                    TypedValue.COMPLEX_UNIT_DIP, SHIELD_ICON_CFR_Y_OFFSET,
+                    resources.displayMetrics
+                ).roundToInt(),
+                Gravity.BOTTOM
+            )
+
+            toolbarShieldIconCfrBinding.closeInfoBanner.setOnClickListener {
+                toolbarShieldIconCfrPopupWindow.dismiss()
+            }
+            toolbarShieldIconCfrPopupWindow.setOnDismissListener {
+                setCfrVisibilityForShieldToolbarIcon()
+            }
+        }
+    }
```

Comment:
Maybe we could move this functionality to helper class, to not overload the `BrowserFragment` class, what do you think?

Reply:
I made the changes .